python/graphstorm/model/utils.py

```diff
@@ -703,8 +703,85 @@ def shuffle_predict(predictions, id_mapping_file, pred_type,
                 len(predictions)).cpu() # predictions are stored in CPU
     return predictions[local_id_mapping]
 
+def shuffle_nids(g, ntype, nids, node_id_mapping_file, rank):
```

Comment:
should we make it stateful? currently, we create a distributed tensor, which is already stateful.

Reply:
As we need rank0 to load the mapping file (we want to avoid loading mapping files every each inferer), we need to create a distributed tensor.

Reply:
i understand. what i mean is that the function is exactly stateful because the distributed tensor is persistent. to avoid creating the distributed tensor and fill the id map to it again and again. i think it's better to make it stateful. in this way, we can control when we should destroy the distributed tensors.

Reply:
Updated